content/troubleshooting/fix-problem-x.md

```diff
@@ -0,0 +1,96 @@
+limit higher. If you're using one of the
+[tugboatqa php images](/setting-up-services/reference/tugboat-images/), use
+something like this in your build script:
+
+`echo "memory_limit = 512M" >> /usr/local/etc/php/conf.d/my-php.ini`
+
+Or you might try something like this in your drupal settings.php:
+
+`if (drupal_is_cli()) { ini_set('memory_limit', '-1');`
+
+## MySQL server has gone away
+
+If you're getting a "MySQL server has gone away" error, you can resolve this by
+increasing the packet size for MySQL:
+
+_BEN CAN WE INSERT AN EXAMPLE FOR THIS?_
+
+## cd isn't working
+
+Each
+[command](/setting-up-services/how-to-set-up-services/leverage-service-commands/)
+is run in its own context, meaning things like `cd` do not "stick" between
+commands. If that behavior is required, include an external script in the git
+repository and call it from the config file.
+
+_BEN CAN WE INSERT AN EXAMPLE OF THIS?_
+
+## Tugboat error messages
+
+- [1064: Command Failed](#1064-command-failed)
+- [1074: Repo configuration does not allow building of pull requests from forks](#1074-repo-configuration-does-not-allow-building-of-pull-requests-from-forks)
+
+### 1064: Command Failed
+
+This error typically occurs when a command in your config file can't be executed
+as written. When a Preview build fails with this error:
+
+1. [Check the Preview logs](../debug-config-file/#how-to-check-the-preview-logs)
+   to see what command was running when the build failed.
+2. If you don't spot an issue with the command,
+   [terminal into the relevant Service](../debug-config-file/#debug-by-terminal-in-tugboat-s-web-ui)
+   and try executing the command directly to get more context about what's
+   happening.
+
+If you're having trouble pinpointing the error, you may need to go back and
+execute your config file commands line-by-line until you find the point at which
+things stop working.
+
+{{% notice tip %}} A few common problems in config files include:
+[`cd` does not "stick" between commands](#cd-isn-t-working) and
+[running a background process "breaks" the build](#running-a-background-process).
+Check out the troubleshooting docs for info on addressing these issues.
+{{% /notice %}}
+
+If you've gone through the debugging process and haven't been able to figure out
+why your command is failing, reach out to us at [help and support](/support/) -
+we're happy to take a look!
+
+### 1074: Repo configuration does not allow building of pull requests from forks
+
+If you're getting this error message, it's because you haven't enabled the
+[repository setting](/setting-up-tugboat/select-repo-settings/) to allow
+building pull requests from forks.
+
+This setting is off by default, but when it's enabled, Tugboat builds Previews
+for pull requests made to the primary repo from forked repositories. Turning on
+this setting should correct this error.
+
+{{% notice warning %}} Any secrets in your Preview will be accessible by the
```

Comment:
Let's be a little more verbose about the dangers here. I'll let you be the wordsmith @dacharyc, but the points to get across are:

- This is a dangerous option to enable for public repositories
- Anyone that can submit a pull request from a fork can
  - extract configured repo environment variables
  - extract the private ssh key generated for the repo
    - this is really only a problem if the repo ssh key is added to some other service, like to pull a database or checkout a related git repo, etc
    - If the repo key is allowed to push changes to a git repo, for example, a malicious user could really screw up that repo